lib/jmespath/nodes/function.rb

```diff
@@ -492,7 +509,7 @@ module CompareBy
 
       def compare_by(mode, *args)
         if args.count == 2
-          values = args[0]
+          values = args[0].to_ary
```

Comment:
this one should convert #to_ary after the check `get_type(values) == ARRAY_TYPE`